src/main/java/hudson/remoting/Launcher.java

```diff
@@ -539,6 +543,11 @@ private SSLSocketFactory getSSLSocketFactory()
                     throw x;
                 } else
                     throw e;
+            } catch (FileNotFoundException e) {
+                System.err.println("Failing to obtain "+slaveJnlpURL);
```

Comment:
Yeah, there is direct usage of System I/O below as well :( Maybe needs refactoring (not in this PR)